### PR TITLE
Migrate TYPO3_DB to DBAL

### PR DESCRIPTION
**Pull request by Nostrabramus**
Migrate all the queries from TYPO3_DB to DBAL

### CODE REVIEW

File: Classes/DirectMailUtility.php [hunk 1]
```diff
@@ -211,79 +227,128 @@ public static function getIdList($table, $pidList, $groupUid, $cat)
...
-                        BackendUtility::deleteClause($switchTable) .
-                        $addWhere,
-                    $switchTable . '.uid, ' . $switchTable . '.email'
-                );
+
+                $res = $queryBuilder
+                    ->selectLiteral( 'DISTINCT ' . $switchTable . '.uid', $switchTable . '.email')
+                    ->from($switchTable)
+                    ->andWhere(
+                        $queryBuilder->expr()->andx()
```
Comment (kartolo) [inline Classes/DirectMailUtility.php]: shouldn't it be andX()? capitalize X

File: .gitignore [hunk 2]
```diff
@@ -1,16 +0,0 @@
-########################
```
Comment (kartolo) [inline .gitignore]: please don't remove this file

File: Classes/Hooks/JumpurlController.php [hunk 3]
```diff
@@ -36,7 +37,7 @@ class JumpurlController
      */
     public function preprocessRequest($parameter, $parentObject)
     {
-        $db = $this->getDatabaseConnection();
+        //$db = $this->getDatabaseConnection();
```
Comment (kartolo) [inline Classes/Hooks/JumpurlController.php]: you could just remove this

File: Classes/Hooks/JumpurlController.php [hunk 4]
```diff
@@ -132,7 +136,7 @@ public function preprocessRequest($parameter, $parentObject)
                         }
                     }
                 }
-                $db->sql_free_result($resMailing);
+                //$db->sql_free_result($resMailing);
```
Comment (kartolo) [inline Classes/Hooks/JumpurlController.php]: you could just remove this

File: Classes/Module/Statistics.php [hunk 5]
```diff
@@ -1599,6 +1733,7 @@ public function makeStatTempTableContent(array $mrow)
                     break;
                 case '1':
                     // treat html links like plain text
+                    break;
```
Comment (kartolo) [inline Classes/Module/Statistics.php]: there must be no break for "1". it's intended, that response_type 1 and 2 will be handled the same way

File: Classes/Module/Statistics.php [hunk 6]
```diff
@@ -1684,24 +1823,42 @@ public function storeRecRec(array $recRec)
...
+                ->from($queryArray[2])
+                ->add('where',$queryArray[3])
+                ->groupBy($queryArray[4])
+                ->orderBy($queryArray[5])
+                ->execute()
+                ->fetchAll();
+        }
+
+
+       /*questa funzione viene chiamata per cambiare la key 'COUNT(*)' in 'counter'*/
```
Comment (kartolo) [inline Classes/Module/Statistics.php]: please in english.

File: Classes/Module/Statistics.php [hunk 7]
```diff
@@ -1850,4 +2016,21 @@ protected function getLanguageService()
     {
         return $GLOBALS['LANG'];
     }
+
+
```
Comment (kartolo) [inline Classes/Module/Statistics.php]: please add some PHPDoc for this function

File: Classes/DirectMailUtility.php (hunk 1 above)
Comment (Nostrabramus) [reply Classes/DirectMailUtility.php]: you're right, 
just done

File: .gitignore (hunk 2 above)
Comment (Nostrabramus) [reply .gitignore]: ok

File: Classes/Hooks/JumpurlController.php (hunk 4 above)
Comment (Nostrabramus) [reply Classes/Hooks/JumpurlController.php]: ops, I forgot this
done

File: Classes/Module/Statistics.php (hunk 6 above)
Comment (Nostrabramus) [reply Classes/Module/Statistics.php]: just done

File: Classes/Hooks/JumpurlController.php (hunk 3 above)
Comment (Nostrabramus) [reply Classes/Hooks/JumpurlController.php]: ops,
just done

File: Classes/Module/Statistics.php (hunk 7 above)
Comment (Nostrabramus) [reply Classes/Module/Statistics.php]: This function was already present in the original code, it's not our code change

**Pull request merged**